### PR TITLE
Add manifest for io.github.swanux.hbud

### PR DESCRIPTION
**Pull request by swanux**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.github.swanux.hbud.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      - mkdir -p /app/bin
+      - mkdir -p /app/share/appdata
+      - mkdir -p /app/share/icons/hicolor/128x128/apps
+      - cp -r DEV_FILES/hbud /app/share/hbud/.
+      - cp -r DEV_FILES/modules /app/share/hbud/. 
+      - cp DEV_FILES/io.github.swanux.hbud.desktop /app/share/applications/.
+      - cp DEV_FILES/io.github.swanux.hbud /app/share/hbud/.
+      - ln -s /app/share/hbud/io.github.swanux.hbud /app/bin/.
+      - cp DEV_FILES/io.github.swanux.hbud.appdata.xml /app/share/appdata/.
+      - cp DEV_FILES/hbud/icons/io.github.swanux.hbud.svg /app/share/icons/hicolor/128x128/apps/.
```
Comment (hfiguiere) [inline io.github.swanux.hbud.yml:64]:
```suggestion
      - install -Dm644 -t /app/share/icons/hicolor/scalable/apps/ DEV_FILES/hbud/icons/io.github.swanux.hbud.svg 
```

svg are scalable.

also at that point it's better to use `install`.

File: io.github.swanux.hbud.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  - "/share/gtk-doc"
+  - "/share/locale"
+  - "/share/runtime"
+  - "/share/kde4"
+  - "/share/metainfo"
+  - "/share/vlc"
+  - "/share/icons/hicolor/16x16"
+  - "/share/icons/hicolor/32x32"
+  - "/share/icons/hicolor/48x48"
+  - "/share/icons/hicolor/256x256"
```
Comment (hfiguiere) [inline io.github.swanux.hbud.yml:32]: why are you removing icons?

File: io.github.swanux.hbud.yml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+sdk: org.gnome.Sdk
+command: io.github.swanux.hbud
+
+finish-args:
+  - --share=network
+  - --socket=wayland
+  - --socket=x11
+  - --device=dri
+  - --socket=pulseaudio
+  - --filesystem=host
```
Comment (hfiguiere) [inline io.github.swanux.hbud.yml:13]: why host for a music player?

File: io.github.swanux.hbud.yml [hunk 4]
```diff
@@ -0,0 +1,139 @@
+app-id: io.github.swanux.hbud
+runtime: org.gnome.Platform
+runtime-version: '41'
+sdk: org.gnome.Sdk
+command: io.github.swanux.hbud
+
+finish-args:
+  - --share=network
+  - --socket=wayland
+  - --socket=x11
```
Comment (hfiguiere) [inline io.github.swanux.hbud.yml:10]:
```suggestion
  - --socket=fallback-x11
```

if it actually support wayland?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: io.github.swanux.hbud.yml (hunk 2 above)
Comment (swanux) [reply io.github.swanux.hbud.yml:32]: To free up some space - as I'm only using libVLC as backend. I don't know if there's a proper build flag to omit icons in libVLC.

File: io.github.swanux.hbud.yml (hunk 4 above)
Comment (swanux) [reply io.github.swanux.hbud.yml:10]: Right... It seems that the GTK integration of libVLC video playback (through Gtk.DrawingArea) doesn't support Wayland. - There's an experimental OpenGL method in the unstable 4.0 branch of libVLC, but I'd wait for it to become upstream before implementing. 

So for now, I should omit wayland.

File: io.github.swanux.hbud.yml (hunk 3 above)
Comment (hfiguiere) [reply io.github.swanux.hbud.yml:12]: `xdg-music` is probably more appropriate

Comment (swanux) [reply io.github.swanux.hbud.yml]: Restricted access to `xdg-music` and `xdg-videos`.

Comment (swanux) [reply io.github.swanux.hbud.yml]: On second thought though, I think it's necessary to provide access to at least the whole `home` dir, in order to be able to use this as a default music and video player.

So settling with `filesystem=home` if it's okay.

**Comment by barthalion** [comment]
bot, build io.github.swanux.hbud

**Pull request closed without merging**